fe1-web/docs/README.md

```diff
@@ -0,0 +1,263 @@
+├── __tests__                  # tests
+│
+├── components                 # library of reusable UI component (React-based)
+│
+├── ingestion                  # module to process incoming messages
+│
+├── model                      # module containing the data model & objects
+│    ├── network               # ... of the objects sent over the network
+│    └── objects               # ... of the application's entities
+│
+├── navigation                 # module dealing with the React navigation
+│
+├── network                    # module to network with the backend
+│
+├── parts                      # module containing the application's unique screens
+│
+├── res                        # resources (assets, images, etc.)
+│
+├── store                      # module dealing with application state (Redux-based)
+│    ├── reducers              # module containing the reducers (Redux)
+│    └── stores                # module to access states outside the React environment
+│
+├── styles                     # stylesheets
+└── utils                      # interfaces to system libraries
+     └── __mocks__             # mocks of those libraries, to be used in tests
+```
+
+### Architecture
+
+The PoP TypeScript frontend provides an interface to allow the users to
+interact as part of a Local Autonomous Organization (LAO),
+within which they will acquire Proof-of-Personhood tokens to identify themselves.
+
+Under the hood, the frontend establishes one or more long-lived websocket
+connections with the servers (organizers, witnesses), through which messages
+are sent back and forth using a publish/subscribe pattern.
+
+On a higher level of abstraction, the frontend may publish messages or subscribe
+for messages on a *channel*. You may think of a channel as a topic which
+stores events that occur over time. For instance, every LAO is denoted by a unique
+channel (also called the LAO channel) and contains messages about all events
+that occur within it, for example, the creation of an election, a roll call.
+A channel may have sub channels associated with it. For instance, a
+LAO may have a sub-channel for the elections where all messages associated
+with that election are published. Please refer to
+[Data pipeline architecture](https://docs.google.com/document/d/19r3rP6o8TO-xeZBM0GQzkHYQFSJtWy7UhjLhzzZVry4/edit#heading=h.1h71fzpdznrh)
+for more information.
+
+All the messages are encoded using JSON and are validated using
+[JSON Schemas](https://json-schema.org) as defined in the
+[protocol folder](https://github.com/dedis/student_21_pop/tree/master/protocol).
+
+[Protocol Specifications](https://docs.google.com/document/d/1fyNWSPzLhM6W9V0VTFf2waMLiJGcscy7wa4bQlLkySM)
+also gives an introduction to the different message formats. However, note that
+the [schemas](https://github.com/dedis/student_21_pop/tree/master/protocol) in this
+repository are **always** the source of truth and are more up to date than the Google Doc.
```

Comment:
I've more or less stolen this part from Gaurav, which makes me think it should belong to a dedicated project-wide documentation :)

Reply:
I think this much duplication is fine. More links to follow, more chances to get lost/distracted ;)